cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1872,33 +1883,46 @@ TEST_F(ScalarTemporalTest, TestLocalTimestamp) {
                      times_seconds_precision, timestamp(u), expected_local_kolkata);
     CheckScalarUnary("local_timestamp", timestamp(u, "Pacific/Marquesas"),
                      times_seconds_precision, timestamp(u), expected_local_marquesas);
+    CheckScalarUnary("local_timestamp", timestamp(u, "-09:30"), times_seconds_precision,
+                     timestamp(u), expected_local_marquesas);
   }
 }
 
 TEST_F(ScalarTemporalTest, TestAssumeTimezone) {
   std::string timezone_utc = "UTC";
   std::string timezone_kolkata = "Asia/Kolkata";
   std::string timezone_us_central = "America/Chicago";
+  std::string timezone_tbilisi = "Asia/Tbilisi";
+  std::string timezone_tbilisi_offset = "+04:00";
   const char* times_utc = R"(["1970-01-01T00:00:00", null])";
   const char* times_kolkata = R"(["1970-01-01T05:30:00", null])";
+  const char* times_tbilisi = R"(["1970-01-01T04:00:00", null])";
   const char* times_us_central = R"(["1969-12-31T18:00:00", null])";
   auto options_utc = AssumeTimezoneOptions(timezone_utc);
   auto options_kolkata = AssumeTimezoneOptions(timezone_kolkata);
   auto options_us_central = AssumeTimezoneOptions(timezone_us_central);
+  auto options_tbilisi = AssumeTimezoneOptions(timezone_tbilisi);
+  auto options_tbilisi_offset = AssumeTimezoneOptions(timezone_tbilisi_offset);
   auto options_invalid = AssumeTimezoneOptions("Europe/Brusselsss");
```

Comment:
Can we now test with several invalid timezones, including several offset zones?

Reply:
Added several examples to `TestNonexistentTimezone` test.